### PR TITLE
HealthCheck support: addresses outstanding issues with PR-763

### PR DESCRIPTION
**Pull request by cdancy**
Fixes shortcomings on #763 

@KostyaSha all tests passed here. Let me know how things look. Thanks.

---
This change is [Reviewable](https://reviewable.io/reviews/docker-java/docker-java/805)

### CODE REVIEW

File: src/main/java/com/github/dockerjava/api/command/HealthState.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    public Integer getFailingStreak() {
+        return failingStreak;
+    }
+
+    public List<Log> getLog() {
+        return log;
+    }
+
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    public static class Log {
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/api/command/HealthState.java]: imho move outside as `HealthStateLog` (will be in list near outer used class)

**Comment by KostyaSha** [comment]
We have some model classes that has a lot of nested classes and listing become too long. So i started moving them outside for new things and don't change already existing to have minimal backward compatibility problem.

File: src/main/java/com/github/dockerjava/api/model/ContainerConfig.java [hunk 2]
```diff
@@ -86,6 +86,9 @@
     @JsonProperty("WorkingDir")
     private String workingDir;
 
+    @JsonProperty("Healthcheck")
+    private Healthcheck healthcheck;
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/api/model/ContainerConfig.java]: class `HealthCheck`

Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/api/model/ContainerConfig.java]: field `healthCheck` as classic java naming

**Comment by cdancy** [comment]
@KostyaSha addressed all comments. Please review.

File: src/main/java/com/github/dockerjava/api/command/InspectContainerResponse.java [hunk 3]
```diff
@@ -390,158 +532,25 @@ public String getFinishedAt() {
...
         }
 
-        @Override
-        public String toString() {
-            return ToStringBuilder.reflectionToString(this);
-        }
-    }
-
-    @JsonIgnoreProperties(ignoreUnknown = true)
-    public static class Mount {
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/api/command/InspectContainerResponse.java]: please don't reorder already existing stuff it makes huge diff

File: src/main/java/com/github/dockerjava/api/model/ContainerConfig.java [hunk 4]
```diff
@@ -430,7 +441,7 @@ public boolean equals(Object o) {
     }
 
     @Override
-    public int hashCode() {
-        return HashCodeBuilder.reflectionHashCode(this);
+    public String toString() {
+        return ToStringBuilder.reflectionToString(this);
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/api/model/ContainerConfig.java]: reordering?

**Comment by KostyaSha** [comment]
@cdancy maybe your IDE reformatting code?

**Comment by cdancy** [comment]
@KostyaSha think this reordering was done by the previous dev. Surely not by me. Let me put things back to how they were and get back to you.

**Comment by KostyaSha** [comment]
> Surely not by me.

Sorry, were trying figure out how much nested classes are here.

**Comment by cdancy** [comment]
@KostyaSha nested classes and methods should now be back in place.

**Comment by cdancy** [comment]
@KostyaSha any chance we could get this merged and put out a new release today?

**Comment by KostyaSha** [comment]
fixuped and merged manually into master, but commit somehow lost you... Please check master and i will run release

**Comment by cdancy** [comment]
@KostyaSha awesome and thanks! 

> but commit somehow lost you

No worries here. So long as the functionality is in that's all I care about ;)

**Comment by KostyaSha** [comment]
@cdancy released

**Comment by cdancy** [comment]
Thanks again @KostyaSha!

**Pull request closed without merging**